portafly/src/components/data-list/BulkSelectorWidget.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react'
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownToggleCheckbox,
+  DropdownItem
+} from '@patternfly/react-core'
+import { useTranslation } from 'i18n/useTranslation'
+import { useDataListTable, useDataListPagination } from 'components/data-list'
+
+interface Props {
+  filteredRows: any[]
+}
+
+const BulkSelectorWidget: React.FunctionComponent<Props> = ({
+  filteredRows
+}) => {
+  const { t } = useTranslation('accounts')
+  const { selectPage, selectAll, selectedRows } = useDataListTable()
+
+  const [isOpen, setIsOpen] = useState(false)
+
+  const selectedCount = selectedRows.length
+
+  const { startIdx, endIdx } = useDataListPagination()
+  const visibleRows = filteredRows.slice(startIdx, endIdx)
+  const pageCount = visibleRows.length
+  const allCount = filteredRows.length
+
+  const onSelect = () => {
+    setIsOpen((current) => !current)
+  }
+
+  // FIXME: Setting isChecked null creates an error in the console for uncontrolled React prop
+  let isChecked: boolean | null
+  if (selectedCount === 0) {
+    isChecked = false
+  } else if (selectedCount === allCount) {
+    isChecked = true
+  } else {
+    isChecked = null
+  }
```

Comment:
Something like the following might be a bit better:

```suggestion
  const isChecked = (selectedCount === 0) ? false : (selectedCount === allCount) ? true : null 
```

Reply:
🙄 🤣

Reply:
There's an Eslint rule preventing us from nesting ternary operators... which I agree with, it can lead to errors or very hard to read sentences. Since the file is small enough it doesn't hurt to have some verbose if/else blocks, I think. On the other hand your suggestion is not so complex. Hopefully @damianpm can give us the casting vote.

Reply:
I know it's not super nice either, but it could be crafted even as a fn if you want...thing is, I'd rather have a nested ternary instead of a variable that changes depending on the result of a nested if-else.